backend/app/mcp_server/tools/knowledge.py

```
@@ -0,0 +1,770 @@
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+            try:
+                scraped = loop.run_until_complete(scraper.scrape_url(url))
+            finally:
+                loop.close()
+
+            if not scraped or not scraped.content:
+                return {"error": f"Failed to scrape content from {url}"}
+
+            content = scraped.content
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)